Firestore/core/test/unit/util/byte_stream_test.cc

```diff
@@ -70,15 +70,88 @@ TEST_P(ByteStreamTest, ReadsEmptyStringStream) {
   EXPECT_TRUE(result.eof());
 }
 
-TEST_P(ByteStreamTest, ReadUtilReadsDelimString) {
+TEST_P(ByteStreamTest, ReadsEdgeCaseSizes) {
+  {
+    auto stream = stream_factory_->CreateByteStream("0123456");
+    auto result = stream->Read(6);
+    EXPECT_EQ(result.ValueOrDie(), "012345");
+    EXPECT_FALSE(result.eof());
+  }
+  {
+    auto stream = stream_factory_->CreateByteStream("0123456");
+    auto result = stream->Read(7);
+    EXPECT_EQ(result.ValueOrDie(), "0123456");
+    EXPECT_TRUE(result.eof());
+  }
+  {
+    auto stream = stream_factory_->CreateByteStream("0123456");
+    auto result = stream->Read(8);
+    EXPECT_EQ(result.ValueOrDie(), "0123456");
+    EXPECT_TRUE(result.eof());
+  }
+  {
+    auto stream = stream_factory_->CreateByteStream("0123456");
+    auto result = stream->ReadUntil('a', 6);
+    EXPECT_EQ(result.ValueOrDie(), "012345");
+    EXPECT_FALSE(result.eof());
+  }
+  {
+    auto stream = stream_factory_->CreateByteStream("0123456");
+    auto result = stream->ReadUntil('a', 7);
+    EXPECT_EQ(result.ValueOrDie(), "0123456");
+    EXPECT_TRUE(result.eof());
+  }
+  {
+    auto stream = stream_factory_->CreateByteStream("0123456");
+    auto result = stream->ReadUntil('a', 8);
+    EXPECT_EQ(result.ValueOrDie(), "0123456");
+    EXPECT_TRUE(result.eof());
+  }
+}
+
+TEST_P(ByteStreamTest, ReadsZeroSizes) {
+  auto stream = stream_factory_->CreateByteStream("0123456");
+  auto result = stream->Read(0);
```

Comment:
Nit: also try with `ReadUntil`?

Reply:
Done. This uncovered a bug, as it was assumed `ReadUntil` always start with empty `buffer_`.

This changed after the peek implementation, however.